### PR TITLE
Initial manifest for org.openscad.OpenSCAD

### PR DESCRIPTION
**Pull request by cosimoc**

### CODE REVIEW

File: org.openscad.OpenSCAD.json [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+        "/share/man",
+        "*.a"
+    ],
+    "modules": [
+        {
+            "name": "boost",
+            "buildsystem": "simple",
+            "build-commands": [
+                "./bootstrap.sh --with-libraries=filesystem,program_options,regex,thread",
+                "./b2 -j `nproc` install --prefix=/app"
```
Comment (barthalion) [inline org.openscad.OpenSCAD.json]: Use `$FLATPAK_BUILDER_N_JOBS` instead of nproc.

**Comment by barthalion** [comment]
bot, build org.openscad.OpenSCAD

**Comment by barthalion** [comment]
Have you tried contacting upstream if they would like to help?

File: org.openscad.OpenSCAD.json [hunk 2]
```diff
@@ -0,0 +1,164 @@
+{
+    "app-id": "org.openscad.OpenSCAD",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.10",
```
Comment (TingPing) [inline org.openscad.OpenSCAD.json]: Use 5.9 or 5.11

File: org.openscad.OpenSCAD.json [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+                },
+                {
+                    "type": "patch",
+                    "path": "openscad-libscintilla.patch"
+                }
+            ],
+            "post-install": [
+                "mkdir -p /app/share/icons/hicolor/512x512/apps",
+                "mv /app/share/pixmaps/openscad.png /app/share/icons/hicolor/512x512/apps",
+                "mv /app/share/mime/packages/openscad.xml /app/share/mime/packages/org.openscad.OpenSCAD.xml"
```
Comment (TingPing) [inline org.openscad.OpenSCAD.json]: I've not looked at the contents but usually you need to `sed` the icon name with the correct one.

File: org.openscad.OpenSCAD.json [hunk 4]
```diff
@@ -0,0 +1,164 @@
+{
+    "app-id": "org.openscad.OpenSCAD",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.10",
+    "sdk": "org.kde.Sdk",
+    "command": "openscad",
+    "rename-appdata-file": "openscad.appdata.xml",
```
Comment (TingPing) [inline org.openscad.OpenSCAD.json]: Upstreams appstream file is missing info and will need to be improved:

- Missing license
- Missing releases
- Old format

File: org.openscad.OpenSCAD.json (hunk 3 above)
Comment (cosimoc) [reply org.openscad.OpenSCAD.json]: Ah yeah... reminds me of https://github.com/flatpak/flatpak-builder/issues/1 !

**Comment by cosimoc** [comment]
@TingPing Thanks for the review. I backported some upstream updates to the Appdata file, and opened another PR to add OARS and release tags. 
I also addressed your other comments.

**Comment by cosimoc** [comment]
@kintel I created a flatpak build manifest for OpenSCAD to be published on Flathub. Since on Flathub we encourage upstreams to maintain their applications independently, are you interested in contributing to keeping it up to date in the future?

**Comment by nedrichards** [comment]
bot, build org.openscad.OpenSCAD

**Comment by cosimoc** [comment]
Seems to have built fine: https://flathub.org/builds/#/builders/2/builds/3133

**Comment by barthalion** [comment]
Repository has been created: https://github.com/flathub/org.openscad.OpenSCAD

Thanks!

**Comment by barthalion** [comment]
@kintel: You should receive an invitation granting you access to the repo above. Make sure to accept it to actually get it.

**Comment by kintel** [comment]
@barthalion I accepted the invite. You may want to invite the rest of the OpenSCAD devs to increase the chances of contributions (https://github.com/orgs/openscad/teams/developers/members).

**Pull request closed without merging**